webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
Pardon my terrible testing, but I did test this on Safari and there it fails:
![screen shot 2016-09-12 at 4 15 57 pm](https://cloud.githubusercontent.com/assets/26739/18451718/0cf79c20-7906-11e6-8e89-2f6e5aae6a04.png)
That means iOS users most likely wouldn't be able to be redirected to where they wanted to go.